### ISSUE TITLE
Need a Maven resovler

### ISSUE THREAD

**Issue by sound2gd** [javaparser/javasymbolsolver#136]
I think we need a resolver that reads pom.xml files and 
automatedly set the src paths and locations for users.

**Comment by ftomassetti** [comment]
Yes, that would be a good idea

**Comment by mlangkabel** [comment]
I've been looking into this and it seems like using the [maven embedder](http://maven.apache.org/ref/3.3.1/maven-embedder/) would be one way to go. The problem here is that the maven embedder has quite some dependencies and until now I thought of the JavaSymbolSolver as a lightweight library.

Another option would be using the maven that is installed on the user's computer.

**Comment by ftomassetti** [comment]
I would say that the constraint on being lightweight is not as strong as for JavaParser. Maybe we could develop a separate plugin for JavaSymbolSolver based on maven embedder.

**Comment by matozoid** [comment]
Well, if you're planning on eventually having symbol solver as a JP maven module, then the constraints go up :)

**Comment by mlangkabel** [comment]
Yes and I think that most of the users like lightweight libraries that don't have too many dependencies. 

However, as we are planning to integrate Maven support into Coati I'm trying to pursue another way and use the Maven command that is already installed on the user's computer. I've implemented this approach in C++ and we'll probably ship a first beta release that includes this feature next week. Then we'll see how things are working out. If this turns out to be a practical approach we can integrate it into JSS as well.

After all we'd just need Maven to fetch a list of source, test and generated code directories that we want the symbol solver to know and figure out a list of dependencies for the project and their location on the used computer. What do you think? Am I missing something here?

**Comment by mlangkabel** [comment]
The [Maven Invoker](https://maven.apache.org/shared/maven-invoker/) is something I just came across. It looks like having much fewer dependencies than the maven embedder!
